### PR TITLE
cmd/devp2p: add eth cmd to send new block announcements

### PR DESCRIPTION
**Pull request by lightclient**
This PR adds a new subcommand to `devp2p`: `eth`. This subcommand has a single command of it's own currently, `new-block` which sends the specified hex as a devp2p block announcement message. We can expand `eth` to support more arbitrary message sending in the future.

```console
$ devp2p eth new-block <enode> <msg>
```

### CODE REVIEW

**Comment by lightclient** [comment]
@fjl PTAL when you have a chance!

File: cmd/devp2p/ethcmd_test.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	if err != nil {
+		tt.Fatalf("unable to get announced block: %v", err)
+	}
+	if hash != got.Hash() {
+		tt.Fatalf("block mismatch (got: %s, want: %s)", hash, got.Hash())
+	}
+}
+
+// runGeth creates and starts a geth node
+func runGeth() (*node.Node, error) {
```
Comment (lightclient) [inline cmd/devp2p/ethcmd_test.go:108]: It's unfortunate this is mostly copied exactly from `cmd/devp2p/internal/ethtest/suite_test.go`. Not sure if there is anything better that could be done.

File: cmd/devp2p/ethcmd.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		ForkID:          chain.ForkID(),
+	}
+	if err := conn.Peer(chain, status); err != nil {
+		return fmt.Errorf("unable to peer with node: %s", err)
+	}
+	// Send new block announcement.
+	code := uint64((ethtest.NewBlock{}).Code())
+	if _, err = conn.Conn.Write(code, msg); err != nil {
+		exit(fmt.Errorf("failed to write to connection: %w", err))
+	}
```
[the remark this answers is not in the document]
Comment (holiman) [reply cmd/devp2p/ethcmd.go:89]: @lightclient want to pursue this?

[the remark this answers is not in the document]
Comment (holiman) [reply cmd/devp2p/ethcmd.go:89]: ..... ping ....

File: cmd/devp2p/ethcmd.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	if err != nil {
+		return fmt.Errorf("error loading chain: %s", err)
+	}
+	conn, err := ethtest.Dial(getNodeArg(ctx))
+	if err != nil {
+		return fmt.Errorf("error dialing peer: %s", err)
+	}
+	// Peer with node.
+	status := &ethtest.Status{
+		ProtocolVersion: 66,
```
Comment (fjl) [inline cmd/devp2p/ethcmd.go:90]: This hard-coded version is going to be a problem. I think it should be passed as a flag.

**Review by fjl — CHANGES_REQUESTED** [review]
The generic command is great, but a cli command with 3 args is not very ergonomic. We already have a command with 3 args, `devp2p rlpx eth-test`, and I could never remember the parameter order... It would be better to handle the args as flags. I suggest that it should be handled like this:

```
devp2p rlpx eth-msg -version 67 -node enr:... <code || msg>
```

i.e. only one positional argument and combined msg code + data.

**Comment by fjl** [comment]
Another note: the original purpose of this command is kind of lost since the merge, because block announcements are not accepted via `eth` anymore. We could probably realize this test with the engine API now.

**Comment by lightclient** [comment]
Yeah that is a good point, I'll see if I can update that hive PR to use the engine API instead.

**Pull request closed without merging**